experimental/examples/bubblebench/cursor-shared-tree/src/appState.ts

```
@@ -0,0 +1,126 @@
+    readonly localClientId: string;
+
+    constructor(
+        private readonly tree: ISharedTree,
+        public width: number,
+        public height: number,
+        numBubbles: number,
+    ) {
+        // Move to root node (which is the Shared AppState Node) and initialize this.clientsSequenceHelper
+        const cursor = tree.forest.allocateCursor();
+        moveToDetachedField(tree.forest, cursor);
+        cursor.enterNode(0);
+        this.clientsSequenceHelper = new SharedTreeSequenceHelper(
+            tree,
+            cursor.buildAnchor(),
+            brand("clients"),
+        );
+        cursor.free();
+
+        // Create the initial JsonableTree for the new local client and insert into the shared tree
+        const initialClientJson = this.makeClientInitialJsonTree(numBubbles);
+        this.localClientId = initialClientJson.clientId;
+        this.clientsSequenceHelper.push(initialClientJson.tree);
+
+        console.log(`created client with id ${this.localClient.clientId} and color ${this.localClient.color}`);
+    }
+
+    public get localClient() {
+        const treeNodes = this.clientsSequenceHelper.getAll();
+        treeNodes.forEach(treeNode => this.shortTermSpawnedAnchors.add(treeNode.anchor));
+
+        const localClientNode = treeNodes
+        .filter(treeNode => treeNode.getFieldValue(Client.clientIdFieldKey) === this.localClientId);
+
+        if (localClientNode.length < 1) {
+           throw new Error('Failed to retreieve local client node');
+        }
+
+        return new Client(this.tree, localClientNode[0].anchor, this.shortTermSpawnedAnchors,);
+    }
+
+    public applyEdits() {
+        this.shortTermSpawnedAnchors.forEach(anchor => {
+            try {
+                this.tree.forest.forgetAnchor(anchor);
+                // console.log('successfully forget anchor')
+            } catch (e) {
+                // console.log("failed to forget anchor.", e)
```

silently suppressing exceptions is not ok. Exceptions should only be caught without rethrowing if the code catching them understands the error and knows the system is in a safe state to continue afterwards.

Also avoid commented out code.

---

Yeah, I honestly don't know if the system is in a safe state when this error occurs but for some reason there are anchors that fail to forget because they are already forgotten,  0x351 /* cannot forget unknown Anchor */.